### ISSUE TITLE
MySQLContainer not starting with empty password

### ISSUE THREAD

**Issue by julius-d** [testcontainers/testcontainers-java#896]
Hi!

I really like testcontainers! Thank you for the great work.
I have a little problem:

When I change the password to empty the startup runs always into a timeout.

```
  @ClassRule
  public static final MySQLContainer<?> mysql = new MySQLContainer(
      MySQLContainer.IMAGE + ":" + "5.6")
      .withUsername("testuser")
      .withPassword("");
```
When reading the log it looks like special handling is needed for empty passwords.

```
 INFO  Container mysql:5.6 is starting: 777559c4179a9ebe9eb910c66c8362a9041a663c96b7b9b779bae36a23355ced
2018-Oct-04 10:01:54,059 [main] 🐳 [mysql:5.6]
 INFO  Waiting for database connection to become available at jdbc:mysql://localhost:32802/test using query 'SELECT 1'
2018-Oct-04 10:03:54,062 [main] 🐳 [mysql:5.6]
 ERROR Could not start container
org.rnorth.ducttape.TimeoutException: org.rnorth.ducttape.TimeoutException: java.util.concurrent.TimeoutException
	at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:53) ~[duct-tape-1.0.7.jar:?]
	at org.testcontainers.containers.JdbcDatabaseContainer.waitUntilContainerStarted(JdbcDatabaseContainer.java:119) ~[jdbc-1.9.1.jar:?]
	at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:292) ~[testcontainers-1.9.1.jar:?]
	at org.testcontainers.containers.GenericContainer.lambda$doStart$0(GenericContainer.java:237) ~[testcontainers-1.9.1.jar:?]
	at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:76) [duct-tape-1.0.7.jar:?]
	at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:235) [testcontainers-1.9.1.jar:?]
	at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:220) [testcontainers-1.9.1.jar:?]
	at org.testcontainers.containers.GenericContainer.starting(GenericContainer.java:738) [testcontainers-1.9.1.jar:?]
	at org.testcontainers.containers.FailureDetectingExternalResource$1.evaluate(FailureDetectingExternalResource.java:29) [testcontainers-1.9.1.jar:?]
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48) [junit-4.12.jar:4.12]
	at org.junit.rules.RunRules.evaluate(RunRules.java:20) [junit-4.12.jar:4.12]
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363) [junit-4.12.jar:4.12]
...
Caused by: org.rnorth.ducttape.TimeoutException: java.util.concurrent.TimeoutException
	at org.rnorth.ducttape.timeouts.Timeouts.callFuture(Timeouts.java:70) ~[duct-tape-1.0.7.jar:?]
	at org.rnorth.ducttape.timeouts.Timeouts.getWithTimeout(Timeouts.java:43) ~[duct-tape-1.0.7.jar:?]
	at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:38) ~[duct-tape-1.0.7.jar:?]
	... 17 more
Caused by: java.util.concurrent.TimeoutException
	at java.util.concurrent.FutureTask.get(FutureTask.java:205) ~[?:1.8.0_162]
	at org.rnorth.ducttape.timeouts.Timeouts.callFuture(Timeouts.java:65) ~[duct-tape-1.0.7.jar:?]
	at org.rnorth.ducttape.timeouts.Timeouts.getWithTimeout(Timeouts.java:43) ~[duct-tape-1.0.7.jar:?]
	at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:38) ~[duct-tape-1.0.7.jar:?]
	... 17 more
2018-Oct-04 10:03:54,063 [main] 🐳 [mysql:5.6]
 ERROR Container log output (if any) will follow:
2018-Oct-04 10:03:54,070 [tc-okhttp-stream-790487766] 🐳 [mysql:5.6]
 INFO  STDERR: error: database is uninitialized and password option is not specified 
2018-Oct-04 10:03:54,070 [tc-okhttp-stream-790487766] 🐳 [mysql:5.6]
 INFO  STDERR:   You need to specify one of MYSQL_ROOT_PASSWORD, MYSQL_ALLOW_EMPTY_PASSWORD and MYSQL_RANDOM_ROOT_PASSWORD

org.testcontainers.containers.ContainerLaunchException: Container startup failed

	at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:242)
	at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:220)
	at org.testcontainers.containers.GenericContainer.starting(GenericContainer.java:738)
	at org.testcontainers.containers.FailureDetectingExternalResource$1.evaluate(FailureDetectingExternalResource.java:29)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.run(SpringJUnit4ClassRunner.java:191)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
Caused by: org.rnorth.ducttape.RetryCountExceededException: Retry limit hit with exception
	at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:83)
	at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:235)
	... 12 more
Caused by: org.testcontainers.containers.ContainerLaunchException: Could not create/start container
	at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:315)
	at org.testcontainers.containers.GenericContainer.lambda$doStart$0(GenericContainer.java:237)
	at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:76)
	... 13 more
Caused by: org.rnorth.ducttape.TimeoutException: org.rnorth.ducttape.TimeoutException: java.util.concurrent.TimeoutException
	at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:53)
	at org.testcontainers.containers.JdbcDatabaseContainer.waitUntilContainerStarted(JdbcDatabaseContainer.java:119)
	at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:292)
	... 15 more
Caused by: org.rnorth.ducttape.TimeoutException: java.util.concurrent.TimeoutException
	at org.rnorth.ducttape.timeouts.Timeouts.callFuture(Timeouts.java:70)
	at org.rnorth.ducttape.timeouts.Timeouts.getWithTimeout(Timeouts.java:43)
	at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:38)
	... 17 more
Caused by: java.util.concurrent.TimeoutException
	at java.util.concurrent.FutureTask.get(FutureTask.java:205)
	at org.rnorth.ducttape.timeouts.Timeouts.callFuture(Timeouts.java:65)
	... 19 more
```

I would be happy, if empty password is supported :)

**Comment by kiview** [comment]
Hi @julius-d, thanks for the suggestion.
We would probably need to add environment variable `MYSQL_ALLOW_EMPTY_PASSWORD` in case empty password is specified.

**Comment by chungngoops** [comment]
@kiview: I really like testcontainers and would like to contribute. Could you please give a suggestion on where I can set the environment variable. I tried to find but have no clue.

**Comment by kiview** [comment]
Hey @chungngoops,
awesome, of course!

So easiest would be to add it to the `withPassword` method in `MySQLContainer`:
https://github.com/testcontainers/testcontainers-java/blob/29570469df18f999645e7d29290fb4ac5dda2672/modules/mysql/src/main/java/org/testcontainers/containers/MySQLContainer.java#L109-L112

You would need to add a call to `addEnv()` method there. Similar to `configure()`:
https://github.com/testcontainers/testcontainers-java/blob/29570469df18f999645e7d29290fb4ac5dda2672/modules/mysql/src/main/java/org/testcontainers/containers/MySQLContainer.java#L38-L47

Alternatively, the check could also happen in the `configure()` method and we add the env variable there conditionally, I think this is nicer.

**Comment by kiview** [comment]
Fixed in #899.
